src/in-depth/signals-channels.rs

```diff
@@ -0,0 +1,31 @@
+use std::time::Duration;
+use crossbeam_channel::{bounded, tick, Receiver, select};
+
+fn ctrl_channel() -> Result<Receiver<()>, ctrlc::Error> {
+    let (sender, receiver) = bounded(100);
+    ctrlc::set_handler(move || {
+        let _ = sender.send(());
+    })?;
+
+    Ok(receiver)
+}
+
+fn main() -> Result<(), exitfailure::ExitFailure> {
+    let ctrls = ctrl_channel()?;
```

Comment:
Nit: maybe this could be renamed to "let ctrlcs = ctrlc_channel()?;"
But I do see "ctrlcs" as kinda ugly. :/

Reply:
Ah, I meant to write `ctrlcs` actually. My fingers probably found it to weird to type, too!

Would calling it `sig_ints` be better or even more confusing?

Reply:
More confusing, IMO :)

Reply:
How about `ctrl_c_events`?

Reply:
Much better, I like it 👍